### ISSUE TITLE
Support WCAG2.1 accessible tool

### ISSUE THREAD

**Issue by Eonasdan** [Eonasdan/tempus-dominus#2390]
Hi @Eonasdan, I have checked docs but couldn't find anything related. Is v6 will support WCAG2.1? In other words will it be an accessible tool?

_Originally posted by @kiliczsh in https://github.com/Eonasdan/tempus-dominus/discussions/2384#discussioncomment-1275978_

**Comment by Eonasdan** [comment]
@kiliczsh The icons have `title` attributes but I haven't done much with accessibility. That's a good idea though.

**Comment by kikonen** [comment]
The thing is that "keyboard accessibility" is mandatory requirement in various places, just due to legalize rrequirements (like ADA in USA).

**Comment by Eonasdan** [comment]
Please check the progress of this feature here: https://stackblitz.com/edit/tempus-dominus-v6-accessibility?file=index.html

I could use some feedback on what's missing and what's working.

**Comment by kikonen** [comment]
I will try to look into it

**Comment by kikonen** [comment]
First notice, if input group with button, is used also to open popup, it must be accessible also via that (FYI, also "button" must be implemented as button, span is fail in ADA tests). Not it seems that when popup is opened via button then keyboard access won't work. Perhaps should move focus back to input field when opening popup (?). I.e. end user most likely tries TAB, not SHIFT+TAB.

"toolbar" at top and "time select" at bottom are not TABable seemingly currently (yep, counted as "fail" in ADA tests).

**Comment by Furgas** [comment]
This could be useful for "how it should work":
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

**Comment by chardin1** [comment]
@Eonasdan What's the status of this feature? I see you marked it as complete in December but the branch doesn't appear to have been merged and released.

**Comment by Eonasdan** [comment]
@chardin1 I'm no longer working on this project. I've moved on to other things. If someone wants to discuss sponsoring this issue, I'd be open to that.

**Comment by MartinRiese** [comment]
@Eonasdan We are interested in sponsoring accessibility improvements. What is the best way to contact you about the specifics? My email address is mriese@dimagi.com, if you want to use email.

**Comment by Eonasdan** [comment]
This issue is being sponsored by Dimagi! It's now in progress.

**Comment by skublik** [comment]
Hello @Eonasdan, when do you expect the function to be completed?

**Comment by Eonasdan** [comment]
For awareness, there's an open PR #2943.

**Comment by jcompagner** [comment]
hmm its a pitty that npm updates this:
"@eonasdan/tempus-dominus": "^6.9.5",

also to 6.10.x

(and not just micro releases so 6.9.11 is allowed but 6.10.x is not)

Now many of or builds that are out in our customers do break because of:


Property 'keyboardNavigation' is missing in type '{ components: { calendar: true; decades: true; year: true; month: true; date: true; clock: true; hours: true; minutes: true; seconds: true; }; calendarWeeks: true; buttons: { today: true; close: true; clear: true; }; inline: false; theme: "light"; }' but required in type '{ keyboardNavigation: boolean; toolbarPlacement?: "top" | "bottom"; components?: { calendar?: boolean; date?: boolean; month?: boolean; year?: boolean; decades?: boolean; clock?: boolean; hours?: boolean; minutes?: boolean; seconds?: boolean; useTwentyfourHour?: boolean; }; ... 8 more ...; placement?: "top" | "botto...'.[0m [1m[35m[plugin angular-compiler][0m

packages/@servoy/bootstrapcomponents/src/calendar/basecalendar.ts:33:8:
[37m      33 â”‚         [32mdisplay[37m:

can't we make that new property optional? i gues that is the problem here? so it doesn't break all existing code?

**Comment by jcompagner** [comment]
hmm it seems all properties in the Options are not mandatory except the new one. i guess that just slipped through it right? it not be mandatory and you can have a nice default:

![Image](https://github.com/user-attachments/assets/32dbfd55-b2fe-44d4-9703-1ca9d26bd430)

**Comment by Eonasdan** [comment]
6.10.1 makes it optional.

I use semantic versioning, since this is a new feature it warrants a minor increase, it's not a bug fix.
